internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_controller.go

```diff
@@ -243,6 +269,16 @@ func (c *Controller) Unlock() bool {
 }
 
 func (c *Controller) run(ctx context.Context, seq runtime.Sequence, phases []runtime.Phase, data interface{}) error {
+	c.ctxMutex.Lock()
```

Comment:
I think we should also add handling for canceled context is between phases,  so if the context is canceled, it shouldn't run the next phase?

my assumption is not all tasks actually use `ctx`, so they might keep running if the context is canceled.

Reply:
(this is not the right place to add this check, I was just trying to add comment somewhere in this file)

Reply:
I think phases loop already handles that properly.
https://github.com/siderolabs/talos/blob/master/internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_controller.go#L296

While tasks loop doesn't and tasks are running in parallel. Can add something there if it makes sense